### ISSUE TITLE
onNotification is not called when a remote notification is pressed on Android

### ISSUE THREAD

**Issue by anisimov74** [zo0r/react-native-push-notification#1378]
Hi,
`onNotification` is not get called when i press a notification. 
The doc https://github.com/zo0r/react-native-push-notification#usage says that 
`onNotification` Called when a remote or local notification is opened or received.

This callback is called twice when a notification is received: 
1. 
`{collapse_key: "XX"
foreground: true
google.delivered_priority: "high"
google.message_id: "0:1587387492329033%3a743ff23a743ff2"
google.original_priority: "high"
google.sent_time: 1587387492305
google.ttl: 2419200
id: "-1393396919"
issueId: "XX"
notification:
body: "XX"
sound: "default"
tag: "XX"
title: "XX"
userInteraction: false}`
2. 
`{foreground: true
id: "-1409941121"
issueId: "XX"
message: "XX"
title: "XX"
userId: "XX"
userInteraction: false}`

Can anybody help me with this problem?

**Comment by gibrandev** [comment]
same here

**Comment by Dallas62** [comment]
Hi!
Can you provide manifest.xml ?
You probably add the GCM receiver which is not needed.

**Comment by anisimov74** [comment]
@Dallas62 here it is
`
  ...
  <uses-permission android:name="android.permission.WAKE_LOCK" />
    <permission
        android:name="${applicationId}.permission.C2D_MESSAGE"
        android:protectionLevel="signature" />
    <uses-permission android:name="${applicationId}.permission.C2D_MESSAGE" />
    

    <uses-permission android:name="android.permission.VIBRATE" />
    <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED"/>


    <application
      android:name=".MainApplication"
      android:allowBackup="true"
      android:usesCleartextTraffic="true"
      android:label="@string/app_name"
      android:icon="@mipmap/yt_logo"
      android:theme="@style/AppTheme">

        <meta-data
            android:name="com.google.firebase.messaging.default_notification_icon"
            android:resource="@mipmap/ic_notification"/>

        <meta-data
            android:name="com.dieam.reactnativepushnotification.notification_color"
            android:resource="@color/notification_color"/>

        <meta-data  android:name="com.dieam.reactnativepushnotification.notification_channel_name"
            android:value="YouTrack"/>
        <meta-data  android:name="com.dieam.reactnativepushnotification.notification_channel_description"
            android:value="YouTrack Issue Tracker"/>

        
        <receiver
            android:name="com.google.android.gms.gcm.GcmReceiver"
            android:exported="true"
            android:permission="com.google.android.c2dm.permission.SEND" >
            <intent-filter>
                <action android:name="com.google.android.c2dm.intent.RECEIVE" />
                <category android:name="com.jetbrains.youtrack.mobile.app" />
            </intent-filter>
        </receiver>
        

        <receiver android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationPublisher" />
        <receiver android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationBootEventReceiver">
            <intent-filter>
                <action android:name="android.intent.action.BOOT_COMPLETED" />
            </intent-filter>
        </receiver>
        <service android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationRegistrationService"/>

        
        <service
            android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationListenerServiceGcm"
            android:exported="false" >
            <intent-filter>
                <action android:name="com.google.android.c2dm.intent.RECEIVE" />
            </intent-filter>
        </service>
        

        
        <service
            android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationListenerService"
            android:exported="false" >
            <intent-filter>
                <action android:name="com.google.firebase.MESSAGING_EVENT" />
            </intent-filter>
        </service>
...
`

**Comment by anisimov74** [comment]
I also unable to get `onNotification` called when the app in a background mode or killed.
`PushNotification.popInitialNotification` always returns null

**Comment by anisimov74** [comment]
@Dallas62 I'm using Firebase to send remote notifications.

**Comment by Dallas62** [comment]
There is conditional configuration in the documentation, look in the comments there is if else, remove those with GCM

**Comment by anisimov74** [comment]
@Dallas62 thank you for reviewing. I changed the content to the following, but it didn't help

```
    <uses-permission android:name="android.permission.VIBRATE" />
    <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED"/>

    <application
      android:name=".MainApplication"
      android:allowBackup="true"
      android:usesCleartextTraffic="true"
      android:label="@string/app_name"
      android:icon="@mipmap/yt_logo"
      android:theme="@style/AppTheme">

        <meta-data
            android:name="com.google.firebase.messaging.default_notification_icon"
            android:resource="@mipmap/ic_notification"/>

        <meta-data
            android:name="com.dieam.reactnativepushnotification.notification_color"
            android:resource="@color/notification_color"/>

        <meta-data  android:name="com.dieam.reactnativepushnotification.notification_channel_name"
            android:value="YouTrack"/>
        <meta-data  android:name="com.dieam.reactnativepushnotification.notification_channel_description"
            android:value="YouTrack Issue"/>

        <receiver android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationPublisher" />

        <receiver android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationBootEventReceiver">
            <intent-filter>
                <action android:name="android.intent.action.BOOT_COMPLETED" />
            </intent-filter>
        </receiver>

        <service android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationRegistrationService"/>

        <service
            android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationListenerService"
            android:exported="false" >
            <intent-filter>
                <action android:name="com.google.firebase.MESSAGING_EVENT" />
            </intent-filter>
        </service>

**Comment by Dallas62** [comment]
Does something changed or nothing changed ?
What is the new comportement ?

**Comment by gibrandev** [comment]
Not working, i tried example project 
`      <meta-data  android:name="com.dieam.reactnativepushnotification.notification_channel_name"
              android:value="Hallopet"/>
      <meta-data  android:name="com.dieam.reactnativepushnotification.notification_channel_description"
                  android:value="Hallopet notification channel"/>

      <receiver
          android:name="com.google.android.gms.gcm.GcmReceiver"
          android:exported="true"
          android:permission="com.google.android.c2dm.permission.SEND" >
          <intent-filter>
              <action android:name="com.google.android.c2dm.intent.RECEIVE" />
              <category android:name="${applicationId}" />
          </intent-filter>
      </receiver>

      <receiver android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationPublisher" />
      <receiver android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationBootEventReceiver">
          <intent-filter>
              <action android:name="android.intent.action.BOOT_COMPLETED" />
          </intent-filter>
      </receiver>
      <service android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationRegistrationService"/>
      <service
          android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationListenerService"
          android:exported="false" >
          <intent-filter>
              <action android:name="com.google.android.c2dm.intent.RECEIVE" />
          </intent-filter>
      </service>
`

**Comment by Dallas62** [comment]
Hi @gibrandev 
Can you be more precise on what is not working ? 
Try with this receiver:
```xml

        <receiver android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationPublisher" />
        <receiver android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationBootEventReceiver">
            <intent-filter>
                <action android:name="android.intent.action.BOOT_COMPLETED" />
            </intent-filter>
        </receiver>
        <service android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationRegistrationService"/>

```

instead of:
```xml
  <receiver
      android:name="com.google.android.gms.gcm.GcmReceiver"
      android:exported="true"
      android:permission="com.google.android.c2dm.permission.SEND" >
      <intent-filter>
          <action android:name="com.google.android.c2dm.intent.RECEIVE" />
          <category android:name="${applicationId}" />
      </intent-filter>
  </receiver>
```
Thanks

**Comment by gibrandev** [comment]
@Dallas62 i am sorry, not working same with this issues

**Comment by anisimov74** [comment]
@Dallas62 the only change is that `onNotification` get called only one time when a notification is received. This callback still is not called when i press the notification.
Tried on a simulator and a real device.
I use Firebase console to send a test notification. Not sure if it's relevant to the problem, but I see no `data: {}` sent by Firebase. I'm using additional options to send custom data.

Could you please point out where i can find the example project. Can't find the ref to it in docs. Thank you!

**Comment by Dallas62** [comment]
@anisimov74 In the `example` folder ;)

**Comment by anisimov74** [comment]
@Dallas62 i tried an example project. It works with local notifications as expected. But it's only about local notification. Right?
I'm using "react": "16.8.3",  "react-native": "0.59.10". May this is the case?

**Comment by gibrandev** [comment]
@anisimov74 Is your case fixed?

**Comment by Dallas62** [comment]
Hi @anisimov74 @gibrandev 
Sorry @581 did not see your reply,
I released a new version, can you test it ?

**Comment by gibrandev** [comment]
@Dallas62 keep not working with this case

**Comment by anisimov74** [comment]
@Dallas62 I have the same problem with the latest version.

**Comment by dhoncrisley** [comment]
same problem here, the `onNotification` is called only on foreground or for local notifications. 
If i press the remote notification nothing happens, only open the app.

**Comment by dhoncrisley** [comment]
I also notice that there is a difference between local and remote notification. When touching the location the app does not restart, it returns from where it left off, on the remote it restarts the app, as if it had opened for the first time. In addition, the floating notification and sounds do not change ...

**Comment by Dallas62** [comment]
Hi,
I don't understand all the issue, does `onNotification` called when the you send a remote notification and the app is closed ?
I tested the `example` project and the `onNotification` is called. Can you check your configuration ?

**Comment by Dallas62** [comment]
I tested with :
- remote and scheduled
- foreground
- background
- closed

`onNotification` always called.

**Comment by dhoncrisley** [comment]
with me fire `onNotification` on:
- Remote on foreground:
- local on foreground;
- local on background;
- local on closed;

not fire on:
- Remote background (Open) - When touch the app appear like reopened
- Remote background(Closed) ;

i've put the `configure()` on entrypoint of app root/index.js;
obs: i'm using Redux and Redux Persist
my configure function called useEffect in a component on root
```javascript
function configure() {
    PushNotification.configure({
      // (optional) Called when Token is generated (iOS and Android)
      onRegister: this.onRegister, //this._onRegister.bind(this),

      // (required) Called when a remote or local notification is opened or received
      onNotification: this.onNotification, //this._onNotification,

      // ANDROID ONLY: FCM Sender ID (optional - not required for local notifications, but is need to receive remote push notifications)
      senderID,//from constant on module scope

      // IOS ONLY (optional): default: all - Permissions to register.
      /* permissions: {
        alert: true,
        badge: true,
        sound: true,
      }, */

      // Should the initial notification be popped automatically
      // default: true
      popInitialNotification: true,

      /**
       * (optional) default: true
       * - Specified if permissions (ios) and token (android and ios) will requested or not,
       * - if not, you must call PushNotificationsHandler.requestPermissions() later
       */
      requestPermissions: true,
    });
  }
```

**Comment by Dallas62** [comment]
Hi @dhoncrisley 
I was speaking about `AndroidManifest.xml`
Your `.configure()` looks good to me.

**Comment by dhoncrisley** [comment]
my AndroidManifest.xml

```xml
<manifest xmlns:android="http://schemas.android.com/apk/res/android" package="com.MYPACKAGE">

  <uses-permission android:name="android.permission.INTERNET" />
  <uses-permission android:name="android.permission.SYSTEM_ALERT_WINDOW"/>
  <uses-permission android:name="android.permission.VIBRATE" />
  <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED"/>
  <uses-permission android:name="android.permission.WAKE_LOCK" />

  <application 
    android:name=".MainApplication" 
    android:label="@string/app_name" 
    android:icon="@mipmap/ic_launcher" 
    android:roundIcon="@mipmap/ic_launcher_round" 
    android:allowBackup="false" 
    android:theme="@style/AppTheme" 
    android:usesCleartextTraffic="true">
      <activity 
        android:name=".MainActivity" 
        android:label="@string/app_name" 
        android:configChanges="keyboard|keyboardHidden|orientation|screenSize" 
        android:windowSoftInputMode="adjustResize"
        android:screenOrientation="portrait">

        <intent-filter>
          <action android:name="android.intent.action.MAIN" />
          <category android:name="android.intent.category.LAUNCHER" />
        </intent-filter>
        <intent-filter>
          <action android:name="android.intent.action.VIEW" />
          <category android:name="android.intent.category.DEFAULT" />
          <category android:name="android.intent.category.BROWSABLE" />
          <data android:scheme="ongold" />
        </intent-filter>
      </activity>
    <activity android:name="com.facebook.react.devsupport.DevSettingsActivity" />

    <receiver android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationPublisher" />
      <receiver android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationBootEventReceiver">
          <intent-filter>
              <action android:name="android.intent.action.BOOT_COMPLETED" />
          </intent-filter>
      </receiver>
      <service android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationRegistrationService"/>

      <service
          android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationListenerService"
          android:exported="false" >
          <intent-filter>
              <action android:name="com.google.firebase.MESSAGING_EVENT" />
          </intent-filter>
      </service>

      <meta-data  android:name="com.dieam.reactnativepushnotification.notification_channel_name"
                  android:value="Compra Daqui!"/>
      <meta-data  android:name="com.dieam.reactnativepushnotification.notification_channel_description"
                  android:value="Canal de Notificações Compra Daqui"/>
      <meta-data  android:name="com.dieam.reactnativepushnotification.notification_color"
                  android:resource="@android:color/white"/>


  

    <provider android:name="androidx.core.content.FileProvider" android:authorities="${applicationId}.provider" android:enabled="true" android:exported="false" android:grantUriPermissions="true">
      <meta-data android:name="android.support.FILE_PROVIDER_PATHS" android:resource="@xml/provider_paths" />
    </provider>


    
  </application>

</manifest>

```

**Comment by MuhammadHassanAmin** [comment]
same problem notification does not call when I click the notification from the tray.

**Comment by Dallas62** [comment]
Did you set `google-services.json` and gradle configuration ?
https://github.com/zo0r/react-native-push-notification#if-you-use-remote-notifications

**Comment by Dallas62** [comment]
Can you try with the version on Master ?

**Comment by gibrandev** [comment]
@Dallas62 i have use example use react native 0.62.2, this issue is work perfect. 
I use the react native 0.61.5 in my case, does it matter?

**Comment by Dallas62** [comment]
Hi @gibrandev the example project, run with the version on master, can you try ?

**Comment by gibrandev** [comment]
@Dallas62 yes i have try example project and work perfect, but in my app used react native 0.61.5 not working as example project.

**Comment by anisimov74** [comment]
@Dallas62 the main difference between an example project and the problem that the example using only local notifications, but we are having problems with remote notifications. In my case it's FCM.

**Comment by Dallas62** [comment]
You can setup Firebase on the example project, it’s the test I make by configuring a Firebase Application on com.example.
I will try later with 0.61.5 but I don’t think this is the problem.

**Comment by gibrandev** [comment]
@Dallas62 I'm currently trying to react native 0.62 to my app at the moment, I want to make sure this has an effect or not.

**Comment by Dallas62** [comment]
Can you try:
```
yarn add zo0r/react-native-push-notification#master
```
or with npm.

**Comment by gibrandev** [comment]
@Dallas62 oke wait, i try this

**Comment by gibrandev** [comment]
@Dallas62 keep not working with master

**Comment by MarcoStb1993** [comment]
I have the same problem with react 16.9, react-native 0.62.0 and your master branch. I receive only data-notifications from firebase v6 and then push a local notification if the app is in background or quit state.
But onNotification only is called when the local Notification is pushed, never when the Notification is opened. 
Below is my AndroidManifest.xml for reference:
```
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
          package="com.sampleapp">

    <uses-permission android:name="android.permission.INTERNET" />
    <uses-permission android:name="android.permission.REQUEST_INSTALL_PACKAGES" />
    <uses-permission android:name="android.permission.DOWNLOAD_WITHOUT_NOTIFICATION" />
    <uses-permission android:name="android.permission.INTERNET" />
    <uses-permission android:name="android.permission.WAKE_LOCK" />
    <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED" />
    <uses-permission android:name="android.permission.VIBRATE" />

    <application
            android:name=".MainApplication"
            android:label="@string/app_name"
            android:icon="@mipmap/ic_launcher"
            android:roundIcon="@mipmap/ic_launcher_round"
            android:allowBackup="false"
            android:theme="@style/AppTheme">
        <activity
                android:name=".MainActivity"
                android:label="@string/app_name"
                android:configChanges="keyboard|keyboardHidden|orientation|screenSize"
                android:windowSoftInputMode="adjustResize"
                android:exported="true">
        </activity>
        <activity
                android:name="com.zoontek.rnbootsplash.RNBootSplashActivity"
                android:theme="@style/BootTheme"> <!-- apply the theme you created at step 3. -->
            <intent-filter>
                <action android:name="android.intent.action.MAIN" />
                <category android:name="android.intent.category.LAUNCHER" />
                <action android:name="android.intent.action.DOWNLOAD_COMPLETE"/>
            </intent-filter>
        </activity>
        <activity android:name="com.facebook.react.devsupport.DevSettingsActivity" />
        <meta-data  android:name="com.dieam.reactnativepushnotification.notification_channel_name"
                    android:value="sampleapp"/>
        <meta-data  android:name="com.dieam.reactnativepushnotification.notification_channel_description"
                    android:value="Test channel for Sample App"/>
        <!-- Change the resource name to your App's accent color - or any other color you want -->
        <meta-data  android:name="com.dieam.reactnativepushnotification.notification_color"
                    android:resource="@color/bootsplash_background"/> <!-- or @android:color/{name} to use a standard color -->
        <receiver android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationPublisher" />
        <receiver android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationBootEventReceiver">
            <intent-filter>
                <action android:name="android.intent.action.BOOT_COMPLETED" />
            </intent-filter>
        </receiver>
        <service android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationRegistrationService"/>
        <service
                android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationListenerService"
                android:exported="false" >
            <intent-filter>
                <action android:name="com.google.firebase.MESSAGING_EVENT" />
            </intent-filter>
        </service>
    </application>
</manifest>
```

**Comment by Dallas62** [comment]
Hum... Maybe a missing dependency...
Can you provide the list of dependencies in `android/app/build.gradle` ?
and also the `package.json` (at least dependencies for firebase and push notifications).

**Comment by Dallas62** [comment]
And if the `'com.google.gms.google-services'` is installed.

**Comment by MarcoStb1993** [comment]
In fact, I forgot to add the firebase-analytics dependency before, but unfortunately nothing changed after adding it.
My app/build.gradle dependencies:
```
dependencies {
    implementation fileTree(dir: "libs", include: ["*.jar"])
    implementation "com.facebook.react:react-native:+"  // From node_modules
    implementation 'androidx.appcompat:appcompat:1.1.0-rc01'
    implementation 'androidx.swiperefreshlayout:swiperefreshlayout:1.1.0-alpha02'
    implementation 'com.google.firebase:firebase-analytics:17.4.0'
```
Also this line is added at the end:
`apply plugin: 'com.google.gms.google-services'`
And the relevant package.json dependencies:
```
"dependencies": {
    "@react-native-firebase/app": "^6.7.1",
    "@react-native-firebase/messaging": "^6.7.1",
    "react": "16.9.0",
    "react-native": "0.62.0",
    "react-native-push-notification": "zo0r/react-native-push-notification#master",
```

How can I check if 'com.google.gms.google-services' are installed?

**Comment by Dallas62** [comment]
I tried without analytics and with rnfirebase/messaging, everything work fine and `onNotification` is called.
Just to be sure, does `onNotification` is called or not? no matter if data or something else is not correct.

**Comment by MarcoStb1993** [comment]
Hm, tried without analytics now and still the same result. `onNotification` is only called when the notification is received but not when it is touched/opened which I want to achieve.

This is the code snippet in which I set the background message handler that expects a message with data containing "path": "Settings":
```
messaging().setBackgroundMessageHandler(async remoteMessage => {
    console.log('Message handled in the background!', remoteMessage);
    if (remoteMessage.hasOwnProperty("data")) {
        if (remoteMessage.data.hasOwnProperty("path")) {
            console.log(remoteMessage.data.path);
            let notif = new NotificationService(function (notification) {
                console.log(notification);
                console.log("push noti received");
                if(notification.userInteraction){
                    console.log("push noti pushed");
                }
            });
        }
    }
});
```

If I understand it correct, a local notification should be pushed when I receive the data-notification from firebase and the notification's content as well as the text "push noti received" written to console. This part works as expected.

The following part does not work: When I touch the notifcation, the console should show the notification's content and the text "push noti pushed" but nothing happens. Touching the notification just opens the app.

**Comment by Dallas62** [comment]
Hum, I don't understand why you put the NotificationService in the BackgroundHandler:
Maybe you should extract this part from the BackgroundHandler.

**Comment by MarcoStb1993** [comment]
I need it to push the local notification with which I hope to get info when the notification is opened. When I just use the firebase messaging listener with title and message-string, I receive notifications for which I do not know how to intercept the user touching the notification.

But maybe, what I'm trying is not making a lot of sense, I don't know. I just couldn't get any other way to work. What I want to achieve is to call a function when a notification from FCM is received (no matter the app state).

**Comment by Dallas62** [comment]
The problem with this kind of implementation, the BackgroundHandler is not instanciate or called at the creation of the App.
You must have an instance of your NotificationServie outside the BackgroundHandler

**Comment by dhoncrisley** [comment]
I tried the sample project and it worked perfectly. So there is really something wrong with my code. After that, I noticed that whenever the app started, I was registering the senderID. So I started to register only once. However it still didn't work, but I realized that when reloading the app, `onNotification` is triggered but without some of the notification information. And it is always triggering the same notification with each reload.

**Comment by Dallas62** [comment]
Hi @dhoncrisley 
This is what I have too, `onNotification` is called but without some information. I was looking at this but I don't know how to get the same result on background/foreground.
I think the App start with data in the `Intent`, so we can only acces to this. An issue comes with data which is not included in `onNotification`. You can test the branch on `zo0r/react-native-push-notification#dev` for this a PR has been merged with a fix.

**Comment by Dallas62** [comment]
But GCM is totally removed and senderID too.

**Comment by MarcoStb1993** [comment]
@Dallas62 : I tried putting the Notification interface into a singleton class object as well and kept it out of the React scope but it nonetheless did not work.
I also went ahead and just copied the example project. There, even without integrating  firebase in the package.json, I do receive remote messages and as it seems the 'onNotification' is called when opening the notification but the alert appearing remains empty.

**Comment by Dallas62** [comment]
@MarcoStb1993 
Yes this this what I tried to explain in :

> I think the App start with data in the Intent, so we can only access to this. An issue comes with data which is not included in onNotification

There is only the data part loaded (fixed on `dev` branch) and the notification part is not loaded, send this kind of notification (from FCM and on the example application):

<img width="711" alt="Capture d’écran 2020-04-28 à 18 08 10" src="https://user-images.githubusercontent.com/3429329/80510694-619ec800-897b-11ea-8709-bb5e921fa3a3.png">

`body` is renamed to `message` when in notification and not from data...
From [`example/App.js`](https://github.com/zo0r/react-native-push-notification/blob/master/example/App.js#L119) we can see what is displayed
```js
    Alert.alert(notif.title, notif.message);
```


I know there is some issues in this library, and I try to address them, but I need to know exactly what happen, this is why I asked from `onNotification`'s comportement 😉

**Comment by Dallas62** [comment]
I found this, could you try ?
https://github.com/zo0r/react-native-push-notification/issues/1304#issuecomment-586135377

**Comment by MarcoStb1993** [comment]
Unfortunately, I cannot get the dev-branch to run. The app instantly crashes after start (example).
But on the example's master-branch, I get what I want: Receiving the notification in all app states and firing the `onNotification` event when I open the notification. Since I don't need the title and message inside the app, only having the data is fine for me.  Just as a note: `onNotification` is now only called when I open the notification and not when it is received.

I tried [this ](https://github.com/zo0r/react-native-push-notification/issues/1304#issuecomment-586135377 )as well with no effect. But I will try to remove the firebase dependencies from my app now and just go with this package, since the example showed me, that it can work the way I want (at least on Android, iOS is yet to come ...)

Thanks a lot for your help and maintaining this package!

**Comment by Dallas62** [comment]
The dev branch might crash because you need to remove:
```xml
<service android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationRegistrationService"/>
```
Which is not needed anymore.

**Comment by MarcoStb1993** [comment]
Thanks, that fixed it! Couldn't see what was causing the problem because the app closed instantly, flashing a red screen which I couldn't read in that time.

Using your above suggestion to add title and message to data as well shows them in the alert as you said!

I finally managed to make it run in my own application by initializing  this package outside of the react scope (in index.js before AppRegistry.registerComponent(...);). Before `onNotification` was not called when the app was in background state and the notification was opened. But I suspect that is because I am using [React Navigation](https://reactnavigation.org/) and [Bootsplash](https://github.com/zoontek/react-native-bootsplash).

Also, I need to add `android:launchMode="singleInstance"` to my MainActivity in the AndroidManifest.xml, otherwise opening the app via touching a notification creates multiple instances of the react activity.

**Comment by Dallas62** [comment]
I will add a fake service to avoid crash for the next release 😉 
I will look to release this today.

**Comment by anisimov74** [comment]
@MarcoStb1993 did you solve this problem still using `@react-native-firebase/messaging` in your project?

**Comment by MarcoStb1993** [comment]
@anisimov74 No, I completely removed all firebase dependencies from build.gradle and package.json. Only added what was mentioned in [this package's readme](https://github.com/zo0r/react-native-push-notification#if-you-use-remote-notifications).
Also copied everything into the AndroidManifest.xml that is mentioned [here](https://github.com/zo0r/react-native-push-notification#android-manual-installation)(just the Manifest part). Furthermore, added the following line to the MainActivity-activity in the manifest:
`android:launchMode="singleInstance"`

This is how my **NotificationInterface.js** looks like which implements a [singleton](https://en.wikipedia.org/wiki/Singleton_pattern) class:
```
import PushNotification from 'react-native-push-notification';

export default class NotificationService {
  constructor() {
    this.configure(this.onRegister, this.onNotif);
    this.lastId = 0;
  }

  static instance = null;

  static getInstance() {
    if (this.instance == null) this.instance = new NotificationService();
    return this.instance;
  }

  onRegister = token => {
    console.log(token);
  };

  onNotif = notif => {
    console.log(notif);
  };

  configure(onRegister, onNotification) {
    PushNotification.configure({
      onRegister: onRegister,
      onNotification: onNotification,
      senderID: 'your senderID from firebase console',
      permissions: {
        alert: true,
        badge: true,
        sound: true,
      },
      popInitialNotification: true,
      requestPermissions: true,
    });
  }

  localNotif() {
    this.lastId++;
    PushNotification.localNotification({
      /* Android Only Properties */
      id: '' + this.lastId, // (optional) Valid unique 32 bit integer specified as string. default: Autogenerated Unique ID
      ticker: 'My Notification Ticker', // (optional)
      autoCancel: true, // (optional) default: true
      largeIcon: 'ic_launcher', // (optional) default: "ic_launcher"
      smallIcon: 'ic_notification', // (optional) default: "ic_notification" with fallback for "ic_launcher"
      bigText: 'My big text that will be shown when notification is expanded', // (optional) default: "message" prop
      subText: 'This is a subText', // (optional) default: none
      color: 'red', // (optional) default: system default
      vibrate: true, // (optional) default: true
      vibration: 300, // vibration length in milliseconds, ignored if vibrate=false, default: 1000
      tag: 'some_tag', // (optional) add tag to message
      group: 'group', // (optional) add group to message
      ongoing: false, // (optional) set whether this is an "ongoing" notification

      /* iOS only properties */
      alertAction: 'view', // (optional) default: view
      category: '', // (optional) default: empty string
      userInfo: {}, // (optional) default: {} (using null throws a JSON value '<null>' error)

      /* iOS and Android properties */
      title: 'Local Notification', // (optional)
      message: 'My Notification Message', // (required)
      playSound: false, // (optional) default: true
      soundName: 'default', // (optional) Sound to play when the notification is shown. Value of 'default' plays the default sound. It can be set to a custom sound such as 'android.resource://com.xyz/raw/my_sound'. It will look for the 'my_sound' audio file in 'res/raw' directory and play it. default: 'default' (default sound is played)
      number: 10, // (optional) Valid 32 bit integer specified as string. default: none (Cannot be zero)
      actions: '["Yes", "No"]', // (Android only) See the doc for notification actions to know more
    });
  }

  scheduleNotif() {
    this.lastId++;
    PushNotification.localNotificationSchedule({
      date: new Date(Date.now() + 30 * 1000), // in 30 secs

      /* Android Only Properties */
      id: '' + this.lastId, // (optional) Valid unique 32 bit integer specified as string. default: Autogenerated Unique ID
      ticker: 'My Notification Ticker', // (optional)
      autoCancel: true, // (optional) default: true
      largeIcon: 'ic_launcher', // (optional) default: "ic_launcher"
      smallIcon: 'ic_notification', // (optional) default: "ic_notification" with fallback for "ic_launcher"
      bigText: 'My big text that will be shown when notification is expanded', // (optional) default: "message" prop
      subText: 'This is a subText', // (optional) default: none
      color: 'blue', // (optional) default: system default
      vibrate: true, // (optional) default: true
      vibration: 300, // vibration length in milliseconds, ignored if vibrate=false, default: 1000
      tag: 'some_tag', // (optional) add tag to message
      group: 'group', // (optional) add group to message
      ongoing: false, // (optional) set whether this is an "ongoing" notification

      /* iOS only properties */
      alertAction: 'view', // (optional) default: view
      category: '', // (optional) default: empty string
      userInfo: {}, // (optional) default: {} (using null throws a JSON value '<null>' error)

      /* iOS and Android properties */
      title: 'Scheduled Notification', // (optional)
      message: 'My Notification Message', // (required)
      playSound: true, // (optional) default: true
      soundName: 'default', // (optional) Sound to play when the notification is shown. Value of 'default' plays the default sound. It can be set to a custom sound such as 'android.resource://com.xyz/raw/my_sound'. It will look for the 'my_sound' audio file in 'res/raw' directory and play it. default: 'default' (default sound is played)
    });
  }

  checkPermission(cbk) {
    return PushNotification.checkPermissions(cbk);
  }

  cancelNotif() {
    PushNotification.cancelLocalNotifications({id: '' + this.lastId});
  }

  cancelAll() {
    PushNotification.cancelAllLocalNotifications();
  }
}
```
And here is my **index.js** referncing it (using React Navigation):
```
import 'react-native-gesture-handler';
import {AppRegistry} from 'react-native';
import Routes from './src/config/routes';
import {name as appName} from './app.json';
import NotificationService from './src/libs/NotificationInterface';

NotificationService.getInstance();

AppRegistry.registerComponent(appName, () => Routes);
```

And the important packages in **package.json**:
```
    "react": "16.9.0",
    "react-native": "0.62.0",
    "react-native-push-notification": "git+https://git@github.com/zo0r/react-native-push-notification.git",
```

Hope this helps!

**Comment by anisimov74** [comment]
@MarcoStb1993 thank you for a quick reply! Do notifications comes in a quit and state?

**Comment by MarcoStb1993** [comment]
Yes, `onNotification` is called when the app is active in foreground and when the notification is opened/touched in background and quit state. And notifications appear as soon as the app is not in foreground state.

Firebase worked for me as well for receiving the notifications but I was unable to detect, when they were opened which is why I switched to the above solution.

**Comment by anisimov74** [comment]
@MarcoStb1993 interesting... `onRegister ` method is never called in my case.

**Comment by anisimov74** [comment]
@Dallas62 i guess this is an error in [docs](https://github.com/zo0r/react-native-push-notification#if-you-dont-use-autolink). The code below should be in `android/app/build.gradle`, but not in `android/build.gradle`
```
dependencies {
    ...
    implementation project(':react-native-push-notification')
    ...
 }
```

**Comment by Dallas62** [comment]
I just release version 3.3.0.

**Comment by anisimov74** [comment]
@Dallas62 thank you! In the docs stated that `onRegister` "Called when Token is generated (iOS and Android)". I tried example project and this method never called. Could you please help?

**Comment by gibrandev** [comment]
@Dallas62 in new version 3.3.0, keep not working this issues.
and local notification in onNotification cannot called, but log is working.
here my code

```javascript
    constructor(props) {
        super(props);
        this.state = {
            senderId: SENDER_ID,
        };
        this.notif = new NotifService(
            this.onRegister.bind(this),
            this.onNotif.bind(this),
        );
    }
    onRegister(token) {
        AsyncStorage.setItem('notification:token', token.token);
    }
    onNotif = (notif) => {
        console.log(notif);
        PushNotification.localNotification({
            /* Android Only Properties */
            id: "0", // (optional) Valid unique 32 bit integer specified as string. default: Autogenerated Unique ID
            ticker: "My Notification Ticker", // (optional)
            autoCancel: true, // (optional) default: true
            largeIcon: "ic_launcher", // (optional) default: "ic_launcher"
            smallIcon: "ic_notification", // (optional) default: "ic_notification" with fallback for "ic_launcher"
            bigText: "My big text that will be shown when notification is expanded", // (optional) default: "message" prop
            subText: "This is a subText", // (optional) default: none
            color: "red", // (optional) default: system default
            vibrate: true, // (optional) default: true
            vibration: 300, // vibration length in milliseconds, ignored if vibrate=false, default: 1000
        });
    }
```

**Comment by Dallas62** [comment]
Sorry @gibrandev 
But I don't understand your code, you are calling localNotification when you receive a notification, this doesn't make sense...

**Comment by gibrandev** [comment]
@Dallas62 in foreground notification not showing, but in console showing data.
In background notification showing, but on click notification not trigered. 
I think calling localNotification make it work foreground, here just test.

**Comment by Dallas62** [comment]
Can you provide updated AndroidManifest or a complete project to reproduce your issue ?
Or I will not be able to reproduce your issue...

**Comment by gibrandev** [comment]
@Dallas62 here my complete AndroidManifest
```xml
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
  package="com.dski.doctor">

    <uses-permission android:name="android.permission.INTERNET" />
    <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
    <uses-permission android:name="android.permission.CAMERA" />
    <uses-permission android:name="android.permission.READ_CONTACTS" />
    <uses-permission android:name="android.permission.READ_CALENDAR" />
    <uses-permission android:name="android.permission.READ_PHONE_STATE" />
    <uses-permission android:name="android.permission.WRITE_CALENDAR" />
    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
    <uses-permission android:name="android.permission.RECORD_AUDIO" />
    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
    <uses-permission android:name="android.permission.WRITE_SETTINGS" />
    <uses-permission android:name="android.permission.VIBRATE" />
    <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED"/>
    <uses-permission android:name="android.permission.WAKE_LOCK" />

    <application
      android:largeHeap="true"
      android:usesCleartextTraffic="true"
      android:name=".MainApplication"
      android:label="@string/app_name"
      android:icon="@mipmap/ic_launcher"
      android:roundIcon="@mipmap/ic_launcher_round"
      android:allowBackup="false"
      android:theme="@style/AppTheme">
      <meta-data
        android:name="com.google.firebase.messaging.default_notification_icon"
        android:resource="@drawable/ic_stat_notification" />
      <meta-data
        android:name="com.google.firebase.messaging.default_notification_color"
        android:resource="@color/base_color" />
      <activity
        android:name=".SplashActivity"
        android:theme="@style/SplashTheme"
        android:label="@string/app_name">
        <intent-filter>
            <action android:name="android.intent.action.MAIN" />
            <category android:name="android.intent.category.LAUNCHER" />
        </intent-filter>
      </activity>
      <activity
        android:name="com.theartofdev.edmodo.cropper.CropImageActivity"
        android:theme="@style/Base.Theme.AppCompat">
      </activity>
      <activity
        android:name=".MainActivity"
        android:label="@string/app_name"
        android:launchMode="singleTask"
        android:configChanges="keyboard|keyboardHidden|orientation|screenSize|uiMode"
        android:screenOrientation="portrait"
        android:windowSoftInputMode="adjustResize"
        android:exported="true" />
      <activity android:name="com.facebook.react.devsupport.DevSettingsActivity" />

      <receiver android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationPublisher" />
      <receiver android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationBootEventReceiver">
          <intent-filter>
              <action android:name="android.intent.action.BOOT_COMPLETED" />
          </intent-filter>
      </receiver>
      <service android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationRegistrationService"/>

      <service
          android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationListenerService"
          android:exported="false" >
          <intent-filter>
              <action android:name="com.google.firebase.MESSAGING_EVENT" />
          </intent-filter>
      </service>

      <meta-data  android:name="com.dieam.reactnativepushnotification.notification_channel_name"
                  android:value="Example-Channel"/>
      <meta-data  android:name="com.dieam.reactnativepushnotification.notification_channel_description"
                  android:value="Super channel description"/>
      <meta-data  android:name="com.dieam.reactnativepushnotification.notification_color"
                  android:resource="@android:color/white"/>
                  
    </application>

</manifest>

```

**Comment by Dallas62** [comment]
> @Dallas62 thank you for the new release!
> In the docs stated that `onRegister` "Called when Token is generated (iOS and Android)". I tried also the example project and this method never called. As far as I understand no user action is required to get a device token. Could you please help?

You have two options:
- set `requestPermissions: true,` to true, and it will be triggered automatically
- set `requestPermissions: false,` to false and call `PushNotification.requestPermissions();`

**Comment by Dallas62** [comment]
@gibrandev I'm not sure have multiple activities is good for the project,
Can you provide a minimal project that reproduce the issue ?

**Comment by gibrandev** [comment]
@Dallas62 of course, I will make it ASAP, and give the report.

**Comment by Dallas62** [comment]
@gibrandev Thanks ! 😃

**Comment by gibrandev** [comment]
@Dallas62 i have done testing with small project, work perfect. In my project use a `"react-native-splash-screen": "^3.2.0"` i think this a problem in my case

**Comment by Dallas62** [comment]
@gibrandev I have a project with `react-native-splash-screen` but I don't have a specific activity for the splash-screen, you should check the configuration of `react-native-splash-screen`.

**Comment by gibrandev** [comment]
@Dallas62 my code is work now 😁, my problem in AndroidManifest you can compare with my comment in top. Thank you for your atention. 🙏
```xml
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
  package="com.dski.doctor">

    <uses-permission android:name="android.permission.INTERNET" />
    <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
    <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
    <uses-permission android:name="android.permission.CAMERA" />
    <uses-permission android:name="android.permission.READ_CONTACTS" />
    <uses-permission android:name="android.permission.READ_CALENDAR" />
    <uses-permission android:name="android.permission.READ_PHONE_STATE" />
    <uses-permission android:name="android.permission.WRITE_CALENDAR" />
    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
    <uses-permission android:name="android.permission.RECORD_AUDIO" />
    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
    <uses-permission android:name="android.permission.WRITE_SETTINGS" />
    <uses-permission android:name="android.permission.VIBRATE" />
    <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED"/>
    <uses-permission android:name="android.permission.WAKE_LOCK" />

    <application
      android:largeHeap="true"
      android:usesCleartextTraffic="true"
      android:name=".MainApplication"
      android:label="@string/app_name"
      android:icon="@mipmap/ic_launcher"
      android:roundIcon="@mipmap/ic_launcher_round"
      android:allowBackup="false"
      android:theme="@style/AppTheme">
      <meta-data
        android:name="com.google.firebase.messaging.default_notification_icon"
        android:resource="@drawable/ic_stat_notification" />
      <meta-data
        android:name="com.google.firebase.messaging.default_notification_color"
        android:resource="@color/base_color" />
      <activity
        android:name=".MainActivity"
        android:label="@string/app_name"
        android:configChanges="keyboard|keyboardHidden|orientation|screenSize"
        android:windowSoftInputMode="adjustResize">
        <intent-filter>
            <action android:name="android.intent.action.MAIN" />
            <category android:name="android.intent.category.LAUNCHER" />
        </intent-filter>
      </activity>
      <activity
        android:name="com.theartofdev.edmodo.cropper.CropImageActivity"
        android:theme="@style/Base.Theme.AppCompat">
      </activity>
      <activity android:name="com.facebook.react.devsupport.DevSettingsActivity" />

      <receiver android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationPublisher" />
      <receiver android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationBootEventReceiver">
          <intent-filter>
              <action android:name="android.intent.action.BOOT_COMPLETED" />
          </intent-filter>
      </receiver>
      <service android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationRegistrationService"/>

      <service
          android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationListenerService"
          android:exported="false" >
          <intent-filter>
              <action android:name="com.google.firebase.MESSAGING_EVENT" />
          </intent-filter>
      </service>

      <meta-data  android:name="com.dieam.reactnativepushnotification.notification_channel_name"
                  android:value="Example-Channel"/>
      <meta-data  android:name="com.dieam.reactnativepushnotification.notification_channel_description"
                  android:value="Super channel description"/>
      <meta-data  android:name="com.dieam.reactnativepushnotification.notification_color"
                  android:resource="@android:color/white"/>
    </application>
</manifest>
```

**Comment by Dallas62** [comment]
Nice !

**Comment by Dallas62** [comment]
@anisimov74 @dhoncrisley @gibrandev 
So everything is fine now ? 😃

**Comment by anisimov74** [comment]
@Dallas62 still no `onRegister` callback.

Could you please take a on my configuration?

**build.gradle**:
```
buildscript {
    ext {
      buildToolsVersion = "28.0.3"
      minSdkVersion = 16
      compileSdkVersion = 28
      targetSdkVersion = 28
      supportLibVersion = "28.0.0"

      firebaseVersion = "17.4.0"
      googlePlayServicesVersion = "16.+"
    }
    repositories {
        jcenter()
        google()
    }
    dependencies {
        classpath 'com.android.tools.build:gradle:3.4.0'
        classpath('com.google.gms:google-services:4.3.3')
        // NOTE: Do not place your application dependencies here; they belong
        // in the individual module build.gradle files
    }
}

allprojects {
    repositories {
        mavenLocal()
        google()
        jcenter()
        maven { url "https://www.jitpack.io" }
        maven {
            // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
            url "$rootDir/../node_modules/react-native/android"
        }
    }
}
```

**settings.gradle**:
```
rootProject.name = 'X'
include ':bugsnag-react-native'
project(':bugsnag-react-native').projectDir = new File(rootProject.projectDir, '../node_modules/bugsnag-react-native/android')
include ':react-native-svg'
project(':react-native-svg').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-svg/android')
include ':react-native-vector-icons'
project(':react-native-vector-icons').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-vector-icons/android')
include ':react-native-gesture-handler'
project(':react-native-gesture-handler').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-gesture-handler/android')
include ':@react-native-community_async-storage'
project(':@react-native-community_async-storage').projectDir = new File(rootProject.projectDir, '../node_modules/@react-native-community/async-storage/android')
include ':react-native-camera'
project(':react-native-camera').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-camera/android')
include ':react-native-image-crop-picker'
project(':react-native-image-crop-picker').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-image-crop-picker/android')
include ':react-native-linear-gradient'
project(':react-native-linear-gradient').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-linear-gradient/android')

include ':app'
include ':react-native-eventsource'
project(':react-native-eventsource').projectDir = new File(rootProject.projectDir, '../node_modules/@huston007/react-native-eventsource/android')
include ':react-native-device-info'
project(':react-native-device-info').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-device-info/android')

include ':react-native-push-notification'
project(':react-native-push-notification').projectDir = file('../node_modules/react-native-push-notification/android')
```

**app/build.gradle**:
```
apply plugin: "com.android.application"
apply from: 'versioning.gradle'

project.ext.vectoricons = [
    iconFontNames: [ 'MaterialCommunityIcons.ttf', 'FontAwesome.ttf' ]
]

apply from: "../../node_modules/react-native-vector-icons/fonts.gradle"

import com.android.build.OutputFile


project.ext.react = [
    entryFile: "index.js",
    // whether to bundle JS and assets in debug mode
    bundleInDebug: false,
    // whether to bundle JS and assets in release mode
    bundleInRelease: true,

    // the root of your RN project, i.e. where "package.json" lives
    root: "../../"
]

apply from: "../../node_modules/react-native/react.gradle"

def enableSeparateBuildPerCPUArchitecture = false
def enableProguardInReleaseBuilds = false

android {
    compileSdkVersion rootProject.ext.compileSdkVersion

    compileOptions {
        sourceCompatibility JavaVersion.VERSION_1_8
        targetCompatibility JavaVersion.VERSION_1_8
    }

    defaultConfig {
        applicationId "X"
        minSdkVersion rootProject.ext.minSdkVersion
        targetSdkVersion rootProject.ext.targetSdkVersion
        versionCode buildVersionCode(appVersion)
        versionName buildVersionName(appVersion)
        vectorDrawables.useSupportLibrary = true
        ndk {
            abiFilters "armeabi-v7a", "x86", "arm64-v8a", "x86_64"
        }
        multiDexEnabled true
        missingDimensionStrategy 'react-native-camera', 'general'
    }
    splits {
        abi {
            reset()
            enable enableSeparateBuildPerCPUArchitecture
            universalApk false  // If true, also generate a universal APK
            include "armeabi-v7a", "x86", "arm64-v8a", "x86_64"
        }
    }
    buildTypes {
        release {
            minifyEnabled enableProguardInReleaseBuilds
            proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
        }
    }
    // applicationVariants are e.g. debug, release
    applicationVariants.all { variant ->
        variant.outputs.each { output ->
            // For each separate APK per architecture, set a unique version code as described here:
            // http://tools.android.com/tech-docs/new-build-system/user-guide/apk-splits
            def versionCodes = ["armeabi-v7a":1, "x86":2, "arm64-v8a": 3, "x86_64": 4]
            def abi = output.getFilter(OutputFile.ABI)
            if (abi != null) {  // null for the universal-debug, universal-release variants
                output.versionCodeOverride =
                        versionCodes.get(abi) * 1048576 + defaultConfig.versionCode
            }
        }
    }

    lintOptions {
        checkReleaseBuilds false
        abortOnError false
    }
}

dependencies {
    implementation project(':react-native-push-notification')
    implementation project(':bugsnag-react-native')
    implementation project(':react-native-svg')
    implementation project(':react-native-vector-icons')
    implementation project(':react-native-gesture-handler')
    implementation project(':@react-native-community_async-storage')
    implementation project(':react-native-camera')
    implementation project(':react-native-image-crop-picker')
    implementation project(':react-native-linear-gradient')
    implementation project(':react-native-eventsource')
    implementation project(':react-native-device-info')
    implementation fileTree(dir: "libs", include: ["*.jar"])
    implementation 'com.google.firebase:firebase-analytics:17.4.0'
    implementation 'androidx.appcompat:appcompat:1.0.0'
    implementation 'androidx.legacy:legacy-support-v4:1.0.0'
    implementation 'androidx.media:media:1.0.0'
    implementation "com.facebook.react:react-native:+"  // From node_modules
}

// Run this once to be able to run the application with BUCK
// puts all compile dependencies into folder libs for BUCK to use
task copyDownloadableDepsToLibs(type: Copy) {
  from configurations.compile
  into 'libs'
}

apply plugin: 'com.google.gms.google-services'
```

**Manifest.xml**:
```
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:tools="http://schemas.android.com/tools"
    package="X"
    android:versionCode="1"
    android:versionName="1.0">

    <uses-permission android:name="android.permission.INTERNET" />
    <uses-permission android:name="android.permission.CAMERA" />
    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"/>

    <uses-permission android:name="android.permission.WAKE_LOCK" />
    <uses-permission android:name="android.permission.VIBRATE" />
    <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED"/>

    <uses-feature android:name="android.hardware.camera" android:required="false"/>
    <uses-feature android:name="android.hardware.camera.autofocus" android:required="false"/>

    <uses-sdk android:targetSdkVersion="22"/>

    <application
      android:name=".MainApplication"
      android:allowBackup="true"
      android:usesCleartextTraffic="true"
      android:label="@string/app_name"
      android:icon="@mipmap/yt_logo"
      android:theme="@style/AppTheme">

        <meta-data
            android:name="com.google.firebase.messaging.default_notification_icon"
            android:resource="@drawable/notification_icon" />
        <meta-data
            android:name="com.google.firebase.messaging.default_notification_color"
            android:resource="@color/notification_icon_color" />

      <activity
        android:name=".SplashActivity"
        android:label="@string/app_name"
        android:theme="@style/SplashTheme"
        android:configChanges="keyboard|keyboardHidden|orientation|screenSize"
        android:windowSoftInputMode="adjustResize">
        <intent-filter>
          <action android:name="android.intent.action.MAIN" />
          <category android:name="android.intent.category.LAUNCHER" />
        </intent-filter>
      </activity>

      <activity
        android:name=".MainActivity"
        android:label="@string/app_name"
        android:launchMode="singleTask"
        android:windowSoftInputMode="adjustResize"
        android:configChanges="keyboard|keyboardHidden|orientation|screenSize">
      </activity>

        <activity android:name="com.facebook.react.devsupport.DevSettingsActivity" />

        <receiver android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationPublisher" />
        <receiver android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationBootEventReceiver">
            <intent-filter>
                <action android:name="android.intent.action.BOOT_COMPLETED" />
            </intent-filter>
        </receiver>
        <service android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationRegistrationService"/>

        <service
            android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationListenerService"
            android:exported="false" >
            <intent-filter>
                <action android:name="com.google.firebase.MESSAGING_EVENT" />
            </intent-filter>
        </service>

        <meta-data  android:name="com.dieam.reactnativepushnotification.notification_channel_name"
            android:value="Example-Channel"/>
        <meta-data  android:name="com.dieam.reactnativepushnotification.notification_channel_description"
            android:value="Super channel description"/>
        <meta-data  android:name="com.dieam.reactnativepushnotification.notification_color"
            android:resource="@color/notification_icon_color"/>

    </application>
</manifest>
```

**MainApplication.java**:
```
package X;

import android.app.Application;

import com.facebook.react.ReactApplication;
import com.bugsnag.BugsnagReactNative;
import com.horcrux.svg.SvgPackage;
import com.oblador.vectoricons.VectorIconsPackage;
import com.swmansion.gesturehandler.react.RNGestureHandlerPackage;
import com.reactnativecommunity.asyncstorage.AsyncStoragePackage;
import org.reactnative.camera.RNCameraPackage;
import com.reactnative.ivpusic.imagepicker.PickerPackage;
import com.BV.LinearGradient.LinearGradientPackage;
import com.github.jonnybgod.RNEventSource.RNEventSourcePackage;
import com.facebook.react.ReactNativeHost;
import com.facebook.react.ReactPackage;
import com.facebook.react.shell.MainReactPackage;
import com.facebook.soloader.SoLoader;

import com.learnium.RNDeviceInfo.RNDeviceInfo;

import java.util.Arrays;
import java.util.List;

import com.dieam.reactnativepushnotification.ReactNativePushNotificationPackage;

public class MainApplication extends Application implements ReactApplication {

  private final ReactNativeHost mReactNativeHost = new ReactNativeHost(this) {
    @Override
    public boolean getUseDeveloperSupport() {
      return BuildConfig.DEBUG;
    }

    @Override
    protected List<ReactPackage> getPackages() {
      return Arrays.<ReactPackage>asList(
            new MainReactPackage(),
            BugsnagReactNative.getPackage(),
            new SvgPackage(),
            new VectorIconsPackage(),
            new RNGestureHandlerPackage(),
            new AsyncStoragePackage(),
            new RNCameraPackage(),
            new PickerPackage(),
            new LinearGradientPackage(),
            new RNEventSourcePackage(),
            new RNDeviceInfo(),
            new ReactNativePushNotificationPackage()
      );
    }

    @Override
    protected String getJSMainModuleName() {
      return "index";
    }
  };

  @Override
  public ReactNativeHost getReactNativeHost() {
    return mReactNativeHost;
  }

  @Override
  public void onCreate() {
    super.onCreate();
    SoLoader.init(this, /* native exopackage */ false);
  }
}

```

**Comment by anisimov74** [comment]
@Dallas62 I found this in Logcat logs:
```
2020-04-29 19:42:07.783 1917-1928/? W/ActivityManager: Unable to start service Intent { cmp=com.example/com.dieam.reactnativepushnotification.modules.RNPushNotificationRegistrationService (has extras) } U=0: not found
```

**Comment by Dallas62** [comment]
With the latest version?
This service is not needed in AndroidManifest

**Comment by anisimov74** [comment]
@Dallas62 it's the example app from the repo
`
"@react-native-community/push-notification-ios": "^1.1.1",
    "react": "16.11.0",
    "react-native": "0.62.2",
    "react-native-push-notification": "git+https://git@github.com/zo0r/react-native-push-notification.git"
`

**Comment by Dallas62** [comment]
There is a new release 3.3.0, this should be better

**Comment by anisimov74** [comment]
@Dallas62 I tried it in my app and post my configuration above ^^. I have the same problem :(

**Comment by Dallas62** [comment]
Did you set this:
```js
  requestPermissions: true,
```

**Comment by anisimov74** [comment]
@Dallas62 sure
```
PushNotification.configure({
      onRegister: function(t) {
        console.log('token generated', t);
      },
      onNotification: onNotification,
      senderID: 'XXXXXXXXXXX',
      permissions: {
        alert: true,
        badge: true,
        sound: true,
      },
      popInitialNotification: true,
      requestPermissions: true,
    });
```

**Comment by Dallas62** [comment]
Really strange,
Did you remove:
```xml
<service android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationRegistrationService"/>
```

**Comment by anisimov74** [comment]
@

> Really strange,
> Did you remove:
> 
> ```
> <service android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationRegistrationService"/>
> ```

Updated as described in the doc

```
<receiver android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationPublisher" />
        <receiver android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationBootEventReceiver">
            <intent-filter>
                <action android:name="android.intent.action.BOOT_COMPLETED" />
            </intent-filter>
        </receiver>

        <service
            android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationListenerService"
            android:exported="false" >
            <intent-filter>
                <action android:name="com.google.firebase.MESSAGING_EVENT" />
            </intent-filter>
        </service>

        <meta-data  android:name="com.dieam.reactnativepushnotification.notification_channel_name"
            android:value="X"/>
        <meta-data  android:name="com.dieam.reactnativepushnotification.notification_channel_description"
            android:value="X"/>
        <meta-data  android:name="com.dieam.reactnativepushnotification.notification_color"
            android:resource="@color/notification_icon_color"/>
```

**Comment by Dallas62** [comment]
senderID is not needed anymore but it's not the problem, did you set the google-service.json of Firebase ?
I don't get it, it's like `PushNotification.configure` is not called 😕

**Comment by anisimov74** [comment]
> senderID is not needed anymore but it's not the problem, did you set the google-service.json of Firebase ?
> I don't get it, it's like `PushNotification.configure` is not called 😕

`google-service.json` is set, `PushNotification.configure` is called.
Could you please check the example project. Does it work properly for you?

**Comment by Dallas62** [comment]
<img width="1026" alt="Capture d’écran 2020-04-30 à 10 48 02" src="https://user-images.githubusercontent.com/3429329/80691150-24981a00-8ad0-11ea-9ef3-367712fbc860.png">

- First console.log comes from `PushNotification.configure`.
- Second one from the click on `Request Permissions`.

Here the test commands:

- Uninstall the application from the device.

```bash
# react-native-push-notification on git:master o [10:42:51] 
$ cd example 

# react-native-push-notification/example on git:master o [10:43:04] 
$ rm -rf node_modules 

# react-native-push-notification/example on git:master o [10:43:20] 
$ cat package.json 
{
  "name": "example",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "start": "react-native start",
    "test": "jest",
    "lint": "eslint ."
  },
  "dependencies": {
    "@react-native-community/push-notification-ios": "^1.1.1",
    "react": "16.11.0",
    "react-native": "0.62.2",
    "react-native-push-notification": "git+https://git@github.com/zo0r/react-native-push-notification.git"
  },
  "devDependencies": {
    "@babel/core": "^7.9.0",
    "@babel/runtime": "^7.9.2",
    "@react-native-community/eslint-config": "^1.0.0",
    "babel-jest": "^25.3.0",
    "eslint": "^6.8.0",
    "jest": "^25.3.0",
    "metro-react-native-babel-preset": "^0.59.0",
    "react-test-renderer": "16.11.0"
  },
  "jest": {
    "preset": "react-native"
  }
}

# react-native-push-notification/example on git:master o [10:43:30] 
$ yarn
yarn install v1.12.3
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
warning "react-native > react-devtools-core > ws@7.2.5" has unmet peer dependency "bufferutil@^4.0.1".
warning "react-native > react-devtools-core > ws@7.2.5" has unmet peer dependency "utf-8-validate@^5.0.2".
warning "@react-native-community/eslint-config > @typescript-eslint/eslint-plugin > tsutils@3.17.1" has unmet peer dependency "typescript@>=2.8.0 || >= 3.2.0-dev || >= 3.3.0-dev || >= 3.4.0-dev || >= 3.5.0-dev || >= 3.6.0-dev || >= 3.6.0-beta || >= 3.7.0-dev || >= 3.7.0-beta".
warning "jest > @jest/core > jest-config > jest-environment-jsdom > jsdom@15.2.1" has unmet peer dependency "canvas@^2.5.0".
[4/4] 📃  Building fresh packages...
success Saved lockfile.
✨  Done in 49.21s.

# react-native-push-notification/example on git:master o [10:45:17] 
$ react-native run-android
info Running jetifier to migrate libraries to AndroidX. You can disable it using "--no-jetifier" flag.
Jetifier found 985 file(s) to forward-jetify. Using 16 workers...
info Starting JS server...
info Installing the app...
Starting a Gradle Daemon, 1 incompatible Daemon could not be reused, use --status for details

> Task :react-native-push-notification:compileDebugJavaWithJavac

> Task :app:stripDebugDebugSymbols UP-TO-DATE
Compatible side by side NDK version was not found.

> Task :app:installDebug
10:46:01 V/ddms: execute: running am get-config
10:46:02 V/ddms: execute 'am get-config' on 'emulator-5554' : EOF hit. Read: -1
10:46:02 V/ddms: execute: returning
Installing APK 'app-debug.apk' on 'Pixel_3a_API_29(AVD) - 10' for app:debug
10:46:02 D/app-debug.apk: Uploading app-debug.apk onto device 'emulator-5554'
10:46:02 D/Device: Uploading file onto device 'emulator-5554'
10:46:02 D/ddms: Reading file permision of react-native-push-notification/example/android/app/build/outputs/apk/debug/app-debug.apk as: rw-r--r-
-
10:46:02 V/ddms: execute: running pm install -r -t "/data/local/tmp/app-debug.apk"
10:46:03 V/ddms: execute 'pm install -r -t "/data/local/tmp/app-debug.apk"' on 'emulator-5554' : EOF hit. Read: -1
10:46:03 V/ddms: execute: returning
10:46:03 V/ddms: execute: running rm "/data/local/tmp/app-debug.apk"
10:46:03 V/ddms: execute 'rm "/data/local/tmp/app-debug.apk"' on 'emulator-5554' : EOF hit. Read: -1
10:46:03 V/ddms: execute: returning
Installed on 1 device.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.0.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 18s
48 actionable tasks: 27 executed, 21 up-to-date
info Connecting to the development server...
8081
info Starting the app on "emulator-5554"...
Starting: Intent { cmp=com.example/.MainActivity }

```

**Comment by MarcoStb1993** [comment]
Unfortunately, I am facing the same issue as @anisimov74 after "upgrading" from master to 3.3.1. So I went back to the example and added the *google-services.json* as well as replacing the react-native-push-notification version in *package.json* with "3.3.1". Then I ran npm install and started the app, nothing was logged and no register token is shown. 

I saw that `implementation 'com.google.firebase:firebase-analytics:17.3.0'` was missing in *android/app/build.gradle*. So, I added it. But still the same, the app is not registered with firebase.

**Comment by Dallas62** [comment]
Thanks for the feedback, I think the initial issue is closed, the issue you discribed @MarcoStb1993 looks like:
https://github.com/zo0r/react-native-push-notification/issues/1380
Can you copy you comment on this issue ?
I will look into it ASAP.

**Comment by MarcoStb1993** [comment]
@Dallas62 I opened a [new issue](https://github.com/zo0r/react-native-push-notification/issues/1403)  since I think the one you mentioned describes a different problem. I hope this is ok.
